src/scripts/main.js

```diff
@@ -1 +1,81 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const id = setTimeout(
+    () => reject(new Error('First promise was rejected')),
+    3000,
+  );
+
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      clearTimeout(id);
+      resolve('First promise was resolved');
+    }
+  });
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', (e) => {
+    resolve('Second promise was resolved');
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    resolve('Second promise was resolved');
+  });
+});
+
+const thirdPromise = new Promise((resolve, reject) => {
+  let leftClick = false;
+  let rightClick = false;
+
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      leftClick = true;
+    }
+
+    if (leftClick && rightClick) {
+      resolve('Third promise was resolved');
+    }
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+    rightClick = true;
+
+    if (leftClick && rightClick) {
+      resolve('Third promise was resolved');
```

Comment:
In `thirdPromise`, the promise resolves when both `leftClick` and `rightClick` are true. However, the 'click' event listener does not prevent the default action for the right-click, which might interfere with the intended behavior. Ensure that the right-click event is properly handled to avoid any conflicts.